src/pile.js

```diff
@@ -1032,6 +1032,54 @@ const createPile = (
     setCover(newCover);
   };
 
+  let labelGraphics;
+  let labelTextContainer;
+
+  const drawLabel = (labels, colors, texts) => {
+    if (isPositioning || isScaling) {
+      postPilePositionAnimation.set('drawLabel', () => {
+        drawLabel(labels, colors, texts);
+      });
+      return;
+    }
+
+    if (!labelGraphics || !labelTextContainer) {
+      labelGraphics = new PIXI.Graphics();
+      labelTextContainer = new PIXI.Container();
+      contentGraphics.addChild(labelGraphics);
+      contentGraphics.addChild(labelTextContainer);
+    } else {
+      labelGraphics.clear();
+      labelTextContainer.removeChildren();
+    }
+
+    const { width, height } = contentGraphics;
+
+    if (labels.length) {
```

Comment:
I think it'd be better to check directly at the beginning of the function if there are actually any labels to draw and if not, clear the graphics and exit the function. E.g.

```
if (!labels.length) {
  ... clear graphics
  return;
}
```